### PR TITLE
Fix failing scenarios

### PR DESCRIPTION
**Pull request by scott-silver**
Attempting to fix as many of the currently-failing scenarios as possible.

The only errors that should remain after this PR are the Base deployments' inability to source cbETH.

### CODE REVIEW

File: scenario/constraints/ReservesConstraint.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import { expect } from 'chai';
+import { Requirements } from './Requirements';
+import { exp } from '../../test/helpers';
+import { ComparisonOp, parseAmount, getToTransferAmount } from '../utils';
+
+export class ReservesConstraint<T extends CometContext, R extends Requirements> implements Constraint<T, R> {
+  async solve(requirements: R, initialContext: T) {
+    const reservesRequirement = requirements.reserves;
+    if (reservesRequirement !== undefined) {
+
```
Comment (kevincheng96) [inline scenario/constraints/ReservesConstraint.ts:12]: nit: extra line

File: scenario/constraints/ReservesConstraint.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      const comet = await initialContext.getComet();
```
Comment (kevincheng96) [inline scenario/constraints/ReservesConstraint.ts:13]: I think we should be pulling Comet from the context in the `barelyMeet` function instead.

**Review by kevincheng96 — APPROVED** [review]
Great fixes 🔥

**Comment by kevincheng96** [comment]
Did the liquidation scenarios just fix themselves?

File: scenario/constraints/ReservesConstraint.ts (hunk 1 above)
Comment (scott-silver) [reply scenario/constraints/ReservesConstraint.ts]: removed

File: scenario/constraints/ReservesConstraint.ts (hunk 2 above)
Comment (scott-silver) [reply scenario/constraints/ReservesConstraint.ts]: updated

File: scenario/MainnetBulkerScenario.ts [hunk 3]
```diff
@@ -138,7 +138,7 @@ scenario(
     expectApproximately(
       (await stETH.balanceOf(albert.address)).toBigInt(),
       (await wstETH.getStETHByWstETH(exp(1, 18))).toBigInt(),
-      1n
+      2n
```
Comment (scott-silver) [inline scenario/MainnetBulkerScenario.ts:141]: not sure why this started diverging by 2 wei all of a sudden 🤔

File: scripts/liquidation_bot/liquidateUnderwaterBorrowers.ts [hunk 4]
```diff
@@ -100,7 +102,7 @@ export const flashLoanPools = {
   },
   arbitrum: {
     usdc: {
-      tokenAddress: addresses.arbitrum.USDT,
+      tokenAddress: addresses.arbitrum.USDC, // USDC/USDC_E/.01% pool
```
Comment (scott-silver) [inline scripts/liquidation_bot/liquidateUnderwaterBorrowers.ts:105]: There's more USDC to borrow in the `USDC/USDC_E` pool (~$2M) than in the `USDT/USDC_E` pool (~$750K)

**Comment by scott-silver** [comment]
> Did the liquidation scenarios just fix themselves?

It's been intermittent; added a new Arbitrum flash loan pool which should help

File: scenario/MainnetBulkerScenario.ts (hunk 3 above)
Comment (kevincheng96) [reply scenario/MainnetBulkerScenario.ts:141]: stETH has a 2 wei rounding issue sometimes https://github.com/lidofinance/lido-dao/issues/442

**Review by kevincheng96 — APPROVED** [review]

**Pull request merged**